src/foundry/client/pixi/layers/grid/grid.d.ts

```diff
@@ -1,171 +1,249 @@
 
-  options: BaseGrid.GridOptions;
+export {};
 
+declare global {
   /**
-   * Grid Unit Width
+   * The base grid class.
+   * This double-dips to implement the "gridless" option
    */
-  w: number;
+  class BaseGrid extends PIXI.Container {
+    constructor(options: BaseGrid.GridOptions);
 
-  /**
-   * Grid Unit Height
-   */
-  h: number;
+    options: BaseGrid.GridOptions;
 
-  /**
-   * Highlight active grid spaces
-   */
-  highlight: PIXI.Container;
-
-  /**
-   * Draw the grid. Subclasses are expected to override this method to perform their type-specific drawing logic.
-   * @param preview - Override settings used in place of those saved to the scene data.
-   *                  (default: `{}`)
-   */
-  draw(preview?: BaseGrid.Preview | undefined): this;
-
-  /**
-   * Highlight a grid position for a certain coordinates
-   * @param layer   - The highlight layer to use
-   * @param options - (default: `{}`)
-   */
-  highlightGridPosition(layer: GridHighlight, options?: BaseGrid.HighlightGridPositionOptions): void;
-
-  /**
-   * Given a pair of coordinates (x, y) - return the top-left of the grid square which contains that point
-   * @returns An Array [x, y] of the top-left coordinate of the square which contains (x, y)
-   */
-  getTopLeft(x: number, y: number): PointArray;
+    /**
+     * Grid Unit Width
+     */
+    w: number;
 
-  /**
-   * Given a pair of coordinates (x, y), return the center of the grid square which contains that point
-   * @param x - The x-coordinate
-   * @param y - The y-coordinate
-   * @returns An array [cx, cy] of the central point of the grid space which contains (x, y)
-   */
-  getCenter(x: number, y: number): PointArray;
+    /**
+     * Grid Unit Height
+     */
+    h: number;
 
-  /**
-   * Given a pair of coordinates (x1,y1), return the grid coordinates (x2,y2) which represent the snapped position
-   * Under a "gridless" system, every pixel position is a valid snapping position
-   *
-   * @param x        - The exact target location x
-   * @param y        - The exact target location y
-   * @param interval - An interval of grid spaces at which to snap.
-   *                   At interval=1, snapping occurs at pixel intervals defined by the grid size
-   *                   At interval=2, snapping would occur at the center-points of each grid size
-   *                   At interval=null, no snapping occurs
-   *                   (default: `null`)
-   * @returns An object containing the coordinates of the snapped location
-   */
-  getSnappedPosition(x: number, y: number, interval?: number | null): { x: number; y: number };
+    /**
+     * Returns the class responsible for the implementation of a given grid type.
+     * @param gridType - The grid type.
+     * @returns (typeof BaseGrid) A constructor for a grid of the given type.
+     */
+    static implementationFor(gridType: CONST.GRID_TYPES): new (...args: any[]) => typeof BaseGrid;
```

Comment:
Isn't `BaseGrid` a class already? So `new (...args: any[]) => typeof BaseClass` is a class that constructs a class.

Looking at the source I'm pretty sure this should just be `typeof BaseClass`.